Tasks/Common/ios-signing-common/ios-signing-common.ts

```diff
@@ -529,73 +529,60 @@ export async function getP12SHA1Hash(p12Path: string, p12Pwd: string): Promise<s
     openssl1.arg(['pkcs12', '-in', p12Path, '-nokeys', '-passin', 'pass:' + p12Pwd]);
 
     let openssl2: ToolRunner = tl.tool(opensslPath);
-    openssl2.arg(['x509', '-noout', '-fingerprint']);
+    openssl2.arg(['x509', '-noout', '-fingerprint', '-subject', '-dates']);
     openssl1.pipeExecOutputToTool(openssl2);
 
-    let sha1Hash: string;
-    openssl1.on('stdout', function (data) {
-        if (data) {
-            // find the fingerprint
-            data = data.toString().trim();
-            let fingerprint: string[] = data.match(/SHA1 Fingerprint=.+/g);
-            if (fingerprint && fingerprint[0]) {
-                sha1Hash = fingerprint[0].replace('SHA1 Fingerprint=', '').replace(/:/g, '').trim();
+    let fingerprint: string;
+    let commonName: string;
+    let notBefore: Date;
+    let notAfter: Date;
+
+    function onLine(line: string) {
+        if (line) {
+            const { key, value } = splitIntoKeyValue(line);
+
+            if (key === 'SHA1 Fingerprint') {
+                fingerprint = value.replace(/:/g, '').trim();
+            } else if (key === 'subject') {
+                const matches: string[] = value.match(/\/CN=([^/]+)/);
+                if (matches && matches[1]) {
+                    commonName = matches[1].trim();
+                }
+            } else if (key === 'notBefore') {
+                notBefore = new Date(value);
+            } else if (key === 'notAfter') {
+                notAfter = new Date(value);
             }
         }
-    })
+    }
+
+    // Concat all of stdout to avoid shearing. This can be updated to `openssl1.on('stdline', onLine)` once stdline mocking is available.
+    let output = '';
+    openssl1.on('stdout', (data) => {
+        output = output + data.toString();
```

Comment:
Since we're not actually streaming this data, we might as well use `execSync` instead of `exec` below.  All of the stdout is available as a property on the result.

Reply:
I gave this a shot. `execSync` breaks `openssl1.pipeExecOutputToTool(openssl2)`. @ericsciple Is it possible to use `execSync` and piping together?

We may have mocking support for `stdline` soon: https://github.com/Microsoft/azure-pipelines-task-lib/pull/462.

Reply:
Oh sorry, I didn't see that.  I can't actually tell what `openssl2` is doing here since we don't ever look at its output.  If we still need it, just disregard what I said.